src/FakeItEasy/ValueProducerSignatureHelper.cs

```diff
@@ -22,6 +24,20 @@ internal static void AssertThatValueProducerSignatureSatisfiesCallSignature(Meth
             }
         }
 
+        internal static void AssertThatValuesSatisfyCallSignature(MethodInfo callMethod, object[] values)
+        {
+            if (IsCallSignatureSatisfiedByValues(callMethod, values))
+            {
+                return;
+            }
+
+            var fakeSignature = BuildSignatureDescription(callMethod.GetParameters().Select(p => p.ParameterType));
+            var actionSignature = BuildSignatureDescription(values.Select(v => v?.GetType()));
+
+            throw new FakeConfigurationException(
+                $"The event has the signature ({fakeSignature}), but the provided arguments have types ({actionSignature}).");

```

Comment:
Sadly, sometimes we put `()`s around signatures, and sometimes not.
And while I'm nitpicking, should the exception messages all move to `ExceptionMessages`?
(I know, I'm getting ahead of myself for comments on a proof of concept.)